src/Topshelf/Configuration/HostConfigurators/HostConfigurator.cs

```diff
@@ -116,5 +116,12 @@ public interface HostConfigurator
         /// <param name="name"></param>
         /// <param name="callback"></param>
         void AddCommandLineDefinition(string name, Action<string> callback);
+
+        /// <summary>
+        /// Adds custom exception handler that will be called for any exception caught
+        /// by Topshelf while a service is starting, running or stopping.
+        /// </summary>
+        /// <param name="callback"></param>
+        void AddExceptionHandler(Action<Exception> callback);

```

Comment:
What if this was a generic method that allowed multiple handlers and filtered on type?
Like:
```void AddExceptionHandler<T>(Action<T> callback) where T:Exception;```